### DISCUSSION TITLE
Embed Grafana in Lovelace?

### DISCUSSION THREAD

**Discussion by chicagoandy** [hassio-addons/addon-grafana#232] [Q&A] [answered]
Is there a recommended approach for embedding Grafana in Lovelace?   All avenues I've explored end in a 401 Unauthorized or me opening a new network port, which I don't love.   I had thought Hassio Ingress would handle authentication but that doesn't seem to be happening.    Any tips?

**Comment by frenck** [comment, marked as answer]
Enable direct access by assigning a port to Grafana in the add-on configuration options (Network section) and restart the add-on.

Access Grafana directly by browsing to your instance using the just configured port. These direct URLs can be used to embed.

**Reply by ZuluPilgrim** [reply]
When I assign a port, I can no longer start the Grafana addon.

The log returns
**[10:16:35] INFO: Starting Grafana...
[10:16:35] FATAL: 
[10:16:35] FATAL: SSL has been enabled using the 'ssl' option,
[10:16:35] FATAL: this requires an SSL certificate file which is
[10:16:35] FATAL: configured using the 'certfile' option in the
[10:16:35] FATAL: add-on configuration.
[10:16:35] FATAL: 
[10:16:35] FATAL: Unfortunately, the file specified in the
[10:16:35] FATAL: 'certfile' option does not exist.
[10:16:35] FATAL: 
[10:16:35] FATAL: Please ensure the certificate file exists and
[10:16:35] FATAL: is placed in the '/ssl/' directory.
[10:16:35] FATAL: 
[10:16:35] FATAL: In case you don't have SSL yet, consider getting
[10:16:35] FATAL: an SSL certificate or setting the 'ssl' option`**

I access my home assistant via a separate NginX Proxy Manager instance. As this manages the certificates, I am not sure what to do about the above.

If I disable SSL then I get “Unable to load iframes pointing at websites using http: if Home Assistant is served over https:."

**Comment by vgdh** [comment]
I have the same issue.
I get an 401: Unauthorized error on a first opening lovelace page.
I have to open a grafana panel first and then I have to refresh a lovelace page

**Reply by frenck** [reply]
Its not an issue of this add-on. See above.

**Reply by chicagoandy** [reply]
I am op and my question specifically mentioned not exposing grafana to the Internet.    

It feels like a bug that once you access grafana through supervisor, there are no 401s, but if you access first through Lovelace you get an error.

**Reply by frenck** [reply]
Again, this is not an add-on issue, but an issue with Home Assistant. There is nothing I can do from an add-on perspective. There is a workaround available as described above, if you don't want to run it that way; the story ends 🤷 Sorry m8! Wish I had better news.
